javarosa/src/main/java/org/javarosa/xform/util/CalendarUtils.java

```diff
@@ -359,14 +359,19 @@ public static UniversalDate incrementYear(UniversalDate date) {
                 year,
                 month,
                 day,
-                toMillisFromJavaEpoch(year, month, day, date.millisFromJavaEpoch % UniversalDate.MILLIS_IN_DAY)
+                toMillisFromJavaEpoch(year, month, day)
         );
     }
 
-    public static long toMillisFromJavaEpoch(int year, int month, int day, long millisOffset) {
+    public static long toMillisFromJavaEpoch(int year, int month, int day) {
+        return toMillisFromJavaEpoch(year, month, day, TimeZone.getDefault());
+    }
+
+    public static long toMillisFromJavaEpoch(int year, int month, int day, TimeZone currentTimeZone) {
         int daysFromMinDay = countDaysFromMinDay(year, month, day);
         long millisFromMinDay = daysFromMinDay * UniversalDate.MILLIS_IN_DAY;
-        return millisFromMinDay + MIN_MILLIS_FROM_JAVA_EPOCH + millisOffset;
+        int timezoneOffsetFromGMT = currentTimeZone.getOffset(millisFromMinDay);
+        return millisFromMinDay + MIN_MILLIS_FROM_JAVA_EPOCH - timezoneOffsetFromGMT;
```

Comment:
Definitely just not following this completely, but can you explain why you always want to be subtracting the `timezoneOffsetFromGMT`